core/src/main/java/com/microsoft/applicationinsights/internal/config/TelemetryConfigurationFactory.java

```diff
@@ -119,7 +120,9 @@ public final void initialize(TelemetryConfiguration configuration) {
     private void setMinimumConfiguration(ApplicationInsightsXmlConfiguration userConfiguration, TelemetryConfiguration configuration) {
```

Comment:
Will the heartbeat module always be added? It didn't look like it the way this method is called.

Reply:
@littleaj look into the method `setTelemetryModules()`. It adds HB module explicitly and also look at`setMininmumConfiguration()`. This module would be present by default.

Reply:
@dhaval24 , Ok, I see. 

I did notice that you can reduce some redundancy if you remove the call to `addHeartbeatModule` from `setMinimumConfiguration`. Since `setTelemetryModules` is always called and `setMinimumConfiguration` is only called when the XML config is missing, you don't need the add call in `setMinimumConfiguration`.

Reply:
@littleaj I need to add this for SpringBoot scenarios. Here it is not called.

Reply:
Ahh, I see. I missed the return statements above.

---
In reply to: [181910995](https://github.com/Microsoft/ApplicationInsights-Java/pull/631#discussion_r181910995) [](ancestors = 181910995)